core/src/main/java/com/yahoo/ycsb/workloads/CoreWorkload.java

```diff
@@ -523,7 +500,7 @@ public String buildKeyName(long keynum) {
    * Builds a value for a randomly chosen field.
    */
   private HashMap<String, ByteIterator> buildSingleValue(String key) {
-    HashMap<String, ByteIterator> value = new HashMap<String, ByteIterator>();
+    HashMap<String, ByteIterator> value = new HashMap<>();
```

Comment:
Are we going to require JDK7+ for YCSB? Or should the core still maintain JDK5 or 6 compatibility?